src/features/repeatable.tsx

```diff
@@ -212,14 +221,17 @@ export function createRepeatable<T extends RepeatableOptions>(
                         {currDisplay.showAmount === false ? null : (
                             <div>
                                 <br />
-                                {unref(genericRepeatable.limit) === Decimal.dInf ? (
-                                    <>Amount: {formatWhole(genericRepeatable.amount.value)}</>
-                                ) : (
-                                    <>
-                                        Amount: {formatWhole(genericRepeatable.amount.value)} /{" "}
-                                        {formatWhole(unref(genericRepeatable.limit))}
-                                    </>
-                                )}
+                                joinJSX(
+                                    <>Amount: {formatWhole(genericRepeatable.amount.value)}</>,
+                                    {unref(genericRepeatable.limit) !== Decimal.dInf ? (
+                                        <> / {formatWhole(unref(genericRepeatable.limit))}</>
+                                    ) : undefined},
+                                    {(genericRepeatable as GenericRepeatable & GenericBonusAmountFeature).bonusAmount == null ? null : (
```

Comment:
Seeing this case here also made me notice that currently the type is not set automatically by using the decorator. Would it be possible to have each decorator include a "transformation" type, that takes the previous type and augments/changes it? Then Each of the outputs (like `Repeatable<T>` for this function) could be replaced with something like `ApplyDecorators<Repeatable<T>, D>` where D is the chain of transformation types from the decorators array

Reply:
This is the point where the decorators are a bit annoying, and the current typing system feels frail enough that I'm scared to try adding to it. There is also the problem that the typing needs to come from somewhere, but all the current functionality is through observer calls, where the modifications to the object itself are only really known to the call that did those modifications.

Using these decorators does at least force you to change up the typing of the OptionsFunc you pass in, which is the main reason I have stuff like `GenericEffectFeature` and `BonusAmountFeatureOptions` being exported from the decorators - so while the decorators themselves are actually pretty dumb and don't know anything except what they specifically work on, the input object does at least need to have those values, and it does guide how to manually set the output type.

Reply:
Ok scratch that on them forcing you to have the proper typing on the input function - the typescript version and takeover mode are being weird, I'm going to need to take another proper look at getting the typing functional

Reply:
Taking a better look at this, specifically - this `as GenericRepeatable & GenericBonusAmountFeature` bit here is because it's on the general display function. While the error notification for including a decorator without its corresponding Options type is entirely useless at the moment, fixing those errors does require adding extra typing to the `createFeature` call (`createUpgrade<UpgradeOptions & EffectFeatureOptions>`), which does let those extra values be used inside the `createFeature` call without any additional casting.

Reply:
While I wish it could do the types automatically, with the docs you wrote I think this is fine for now